sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/ExternalCatalog.scala

```diff
@@ -158,7 +173,13 @@ abstract class ExternalCatalog
    * @param table Name of table to alter schema for
    * @param newDataSchema Updated data schema to be used for the table.
    */
-  def alterTableDataSchema(db: String, table: String, newDataSchema: StructType): Unit
+  final def alterTableDataSchema(db: String, table: String, newDataSchema: StructType): Unit = {
+    postToAll(AlterTableSchemaPreEvent(db, table))
```

Comment:
`AlterTableDataSchemaXXX`

Reply:
shall we carry the new schema?

Reply:
For me I think it is not so necessary to carry the new schema, we can query the catalog by `db` and `table` to get this newly set schema.